jgiven-core/src/main/java/com/tngtech/jgiven/format/table/FieldBasedRowFormatter.java

```diff
@@ -47,14 +56,63 @@ public FieldBasedRowFormatter( Class<?> type, String parameterName, Table tableA
     }
 
     @Override
-    public List<String> formatRow( Object object ) {
-        List<Object> allFieldValues = ReflectionUtil.getAllFieldValues( object, fields, "" );
-        for( int i = 0; i < allFieldValues.size(); i++ ) {
-            if( allFieldValues.get( i ) != null ) {
+    public List<String> formatRow(Object object) {
+        ArgumentFormatter formatter;
+
+        List<Object> allFieldValues = ReflectionUtil.getAllFieldValues(object,
+                fields, "");
+
+        List<String> res = Lists.newArrayList();
+
+        for (int i = 0; i < allFieldValues.size(); i++) {
+            Object v = allFieldValues.get(i);
+            Field field = fields.get(i);
+            Format format = formatsByFieldName.get(field.getName());
+
+            if (v != null) {
                 nonNullColumns[i] = true;
             }
+
+            if (format == null) {
+            	formatter = DefaultFormatter.INSTANCE;
+                res.add(formatter.format(v));
+            } else {
+                Class<? extends ArgumentFormatter<?>> clazz = format.value();
+                formatter = ReflectionUtil.newInstance(clazz);
+                res.add(formatter.format(v, format.args()));
+            }
         }
-        return toStringList( allFieldValues );
+
+        return res;
+    }
+
+    private Map<String, Format> retrieveFieldsFormats() {
+        Map<String, Format> inter = Maps.newHashMap();
+        for (Annotation annotation : annotations) {
+            TableFieldsFormats tffs = null;
+
+            Class<? extends Annotation> annotationType = annotation
+                    .annotationType();
+            if (annotationType.isAssignableFrom(TableFieldsFormats.class)) {
+                tffs = (TableFieldsFormats) annotation;
+            } else if (annotationType
+                    .isAnnotationPresent(TableFieldsFormats.class)) {
+                tffs = (TableFieldsFormats) annotationType
+                        .getAnnotation(TableFieldsFormats.class);
+            }
+
+            if (tffs != null) {
+                TableFieldFormat[] tffTab = tffs.value();
+                for (TableFieldFormat tff : tffTab) {
+                    String fieldName = tff.value();
+                    Format format = tff.format();
```

Comment:
a little disadvantage here is that you cannot have custom annotations. This could make this a bit cumbersome to use. It would be great if all ways of formatting could be supported.